docs/content/docs/wip/ingress.md

```diff
@@ -0,0 +1,427 @@
+The OSM ingress doc is a walkthrough of how to expose HTTP and HTTPS routes from outside the cluster to services inside the cluster, and which ingress resources are compatible with OSM v0.8.0. 
+
+## Prerequisites
+- An instance of OSM must be running in the cluster.
+- The service needing to be exposed using Ingress needs to belong to a namespace monitored by OSM. Refer to the [Readme][1] for details.
+- The ingress resource must belong to the same namespace as the backend service.
+- A sidecar must be injected to the pod hosting the service, either using automatic sidecar injection or by manually annotating the pod spec for sidecar injection. Refer to the [Readme][1] for details.
+- An ingress controller must be running in the cluster.
+
+## What is ingress? 
+
+Ingress refers to communication from services outside of your cluster to services inside the cluster, usually via HTTP/HTTPS. In kubernetes, Ingress consists of an Ingress API object and an Ingress Controller. The Kubernetes Ingress API manifest consists of declarations of how clients are routed to a service inside the cluster, and the Ingress controller executes the routing declarations specified in the manifest. 
+
+## Why do you need ingress on your cluster? 
+
+Initially, when you group pods under a service, this service is only accessible inside of the cluster. One option, of course, is to use a load balancer. However, the downside of this approach is that each service would require a new hosted load balancer, meaning more consumption of cloud resources. 
+
+However, Ingress allows you to easily establish rules for traffic routing without creating several load balancers. The Ingress Controller is built using reverse proxies, allowing it to act as a load balancer. Though the Ingress Controller requires a service to expose them to services outside of the cluster, this will be the only entrypoint that outside services need to access all the internal services on your cluster, as the Ingress Controller will redirect this traffic to the internal pods as specified in the Ingress Manifest. 
+
+## What types of ingress are supported by OSM v0.8.0?
+
+Currently, OSM v0.8.0 supports http and single service Ingress. HTTPS ingress support is experimental. OSM supports one way TLS authentication to backend services.
+
+## Ingress controller compatibility
+
+Ingress in OSM is compatible with the following ingress controllers.
+- [Kubernetes Nginx Ingress Controller][2]
+- [Azure Application Gateway Ingress Controller][3]
+- [Gloo API Gateway][5]
+
+Other ingress controllers might also work as long as they use Kubernetes Ingress resource, and allow provisioning a custom root certificate for HTTPS backend server certificate validation while using HTTPS ingress.
+
+## Configuring Ingress
+
+### Enabling Ingress
+Ingress is disabled by default when OSM is installed. However, HTTPS ingress can be enabled by updating the `osm-config` ConfigMap in `osm-controller`'s namespace (`osm-system` by default).
```

Comment:
This statement is incorrect. Ingress is enabled when an ingress resource is applied. By default, HTTP based ingress is provisioned in OSM, which is a global setting. The config map allows globally configuring HTTPS ingress instead when an ingress resource is applied.